### PR TITLE
ISPN-11914 ServerRunMode.FORKED should use by default the server folder

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.redhat.com/browse/ISPN-11914

### CODE REVIEW

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java [hunk 1]
```diff
@@ -37,17 +44,34 @@ public void prepare(String name) {
...
    protected void start(String name, File rootDir, File configurationFile) {
-      String allServerHomes = System.getProperty(TestSystemPropertyNames.INFINISPAN_SERVER_HOME);
-      if (allServerHomes == null) {
-         throw new IllegalArgumentException("You must specify a " + TestSystemPropertyNames.INFINISPAN_SERVER_HOME + " property pointing to a comma-separated list of server homes.");
+      if (configuration.isParallelStartup()) {
+         throw new IllegalStateException("Forked mode doesn't support parallel startup");
       }
-      String[] serverHomes = allServerHomes.replaceAll("\\s+", "").split(",");
-      if (serverHomes.length != configuration.numServers()) {
+      String globalServerHome = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_DIR);
```
Comment (rhusar) [inline server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java:50]: This appears wrong, please see the fix in PR #8378.

Comment (diegolovison) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java:50]: We need to wait for #8378 and then merge this.

**Comment by diegolovison** [comment]
@rhusar it is ready for review. Main changes are:

- TestSystemPropertyNames.INFINISPAN_SERVER_HOME becomes TestSystemPropertyNames.INFINISPAN_TEST_SERVER_DIR. We don't need a new property to say the server path
- You only need to send the server path once. The driver is going to create the folders based on your cluster size. Before this PR you need to send a fixed number of servers comma separated.
- `throw new IllegalStateException("Forked mode doesn't support parallel startup");`

I will suggest to you if possible, run your tests based on my PR.

Thanks

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java (hunk 1 above)
Comment (rhusar) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java]: You can resolve this topic (and still needs a rebase).

**Comment by tristantarrant** [comment]
@diegolovison closing this PR. Please reopen when you are ready to work on it again.

**Pull request closed without merging**